core/runtime/TRTEngine.cpp

```diff
@@ -248,6 +254,43 @@ void TRTEngine::set_profiling_paths() {
   cuda_graph_debug_path = std::filesystem::path{profile_path_prefix + "/" + name + "_cudagraph.dot"}.string();
 }
 
+int64_t TRTEngine::get_weight_streaming_budget_v2() {
+  return cuda_engine->getWeightStreamingBudgetV2();
+}
+
+bool TRTEngine::set_weight_streaming_budget_v2(int64_t budget) {
```

Comment:
Lets make these names more clear to people who have never used TRT before. 

Would something like `TRTEngine::set_device_memory_budget` make sense, does that accurately describe the setting here?

Reply:
I agree with that. There is device_memory_size api in TRT api, it might be confusing.
https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Core/Engine.html

what about to name it to set_weight_streaming_budget()?